karma.conf.js

```diff
@@ -39,22 +39,32 @@ const newWebpackConfig = Object.assign({}, webpackConfigProd, {
     function WebpackWarningPlugin() {
       this.plugin('done', (_stats) => {
         const stats = _stats;
-        let flagError;
-
-        if (stats.compilation.warnings.length) {
-          // Log each of the warnings
-          stats.compilation.warnings.forEach((_warning) => {
-            const warning = _warning.message || _warning || '';
-            if (warning.indexOf('SyntaxError') > -1) {
-              console.log(warning);
-              flagError = true;
-            }
-          });
-
-          // Bail if syntax errors were encountered.
-          if (flagError) {
-            throw new Error('Syntax error encountered, bailing');
+        let flagException = false;
+
+        const loggedExceptions = [
+          'SyntaxError',
+        ];
+
+        function strContainsException(str) {
+          return loggedExceptions.some((substring) => str.includes(substring));
+        }
+
+        ['errors', 'warnings'].forEach((key) => {
+          const loggedStack = stats.compilation[key];
+          if (loggedStack && loggedStack.length) {
+            stats.compilation[key].forEach((item) => {
+              const message = item.message || item || '';
```

Comment:
could `item` really be falsey here? If so, `item.message` is going to throw. If not, then you probably don't need some of those or statements.